engine/docker/docker.go

```diff
@@ -324,18 +324,37 @@ func (e *Docker) create(ctx context.Context, pipelineConfig *spec.PipelineConfig
 		)
 	}
 
+	originalImage := step.Image
+	overriddenImage := originalImage
+
+	// override image registry for internal images
+	// this is short term solution
+	// override if no auth is present
```

Comment:
Instead of original, overridden, can we use more descriptive names to indicate `dockerHubImage`, `garImage` or similar?